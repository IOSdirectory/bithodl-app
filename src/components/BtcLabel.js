import React, { Component } from 'react';
import PropTypes from 'prop-types';
import StyledText from './StyledText';

const UNIT_BTC = 'BTC';
const UNIT_MBTC = 'mBTC';

export default class BtcLabel extends Component {
  _formatBtc(satoshis) {
    const btc = satoshis / 100000000;
    return `${btc} ₿`;
  }

  _formatMBtc(satoshis) {
    const mbtc = satoshis / 100000;
    return `${mbtc} m₿`;
  }

  render() {
    const { amount, unit } = this.props;
    let label;

    switch (unit) {
      case UNIT_MBTC:
        label = this._formatMBtc(amount);
        break;

      case UNIT_BTC:
      default:
        label = this._formatBtc(amount);
    }

    return (
      <StyledText style={this.props.style}>
        {label}
      </StyledText>
    );
  }
}

BtcLabel.propTypes = {
  style: PropTypes.any,
  amount: PropTypes.number.isRequired,
  unit: PropTypes.string
};
