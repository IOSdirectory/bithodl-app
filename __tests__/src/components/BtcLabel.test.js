import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import BtcLabel from '../../../src/components/BtcLabel';

describe('BtcLabel', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <BtcLabel amount={21123456} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('#_formatBtc(satoshis)', () => {
    let btcLabel;

    beforeEach(() => {
      btcLabel = new BtcLabel();
    });

    it('accepts one argument', () => {
      const length = btcLabel._formatBtc.length;
      expect(length).toBe(1);
    });

    it('returns the amount as BTC followed by "₿"', () => {
      const satoshis = 21123456;
      const returnValue = btcLabel._formatBtc(satoshis);

      expect(typeof returnValue).toBe('string');
      expect(returnValue).toBe('0.21123456 ₿');
    });
  });
});
