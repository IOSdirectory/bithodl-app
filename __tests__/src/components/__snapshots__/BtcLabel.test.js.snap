// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtcLabel renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  numberOfLines={undefined}
  style={
    Array [
      Object {
        "color": "#504F63",
        "fontFamily": "Avenir Next",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  0.21123456 â‚¿
</Text>
`;
